src/components/product-switcher/index.tsx

```diff
@@ -0,0 +1,243 @@
+     * The reason we can't focus the first anchor here is because we have the anchor element
+     * is not rendered until after `setIsOpen(true)` is called by the <button>'s onClick (default
+     * behavior of <button> elements).
+     *
+     * Might be possible to do e.preventDefault and then do the focus() call here? Not sure we
+     * need to do that though unless we are very against this approach. I'd rather not prevent
+     * default behavior if we don't have to.
+     */
+    if (!isOpen && (isEnterKey || isSpaceKey)) {
+      shouldFocusFirstAnchor.current = true
+      return
+    }
+
+    if (isEscapeKey) {
+      setIsOpen(false)
+      buttonRef?.current?.focus()
+      return
+    }
+  }
+
+  const renderProductListItem = (
+    product: Product,
+    index: number
+  ): ReactElement => {
+    const isCurrentProduct = product.code === currentProduct.code
+
+    const handleAnchorKeyDown: KeyboardEventHandler<HTMLAnchorElement> = (
+      e
+    ) => {
+      const lastIndex = products.length - 1
+      const isFirstItem = index === 0
+      const isLastItem = index === lastIndex
+      if (!(isFirstItem || isLastItem)) {
+        return
+      }
+
+      const isTabbingForward = !e.shiftKey && e.key === 'Tab'
+      const isTabbingBackward = e.shiftKey && e.key === 'Tab'
+      if (!(isTabbingForward || isTabbingBackward)) {
+        return
+      }
+
+      const anchorElements = getAllAnchorElements()
+      if (isFirstItem && isTabbingBackward) {
+        const lastAnchorElement = anchorElements[lastIndex]
+        lastAnchorElement.focus()
+        e.preventDefault()
+      } else if (isLastItem && isTabbingForward) {
+        const firstAnchorElement = anchorElements[0]
+        firstAnchorElement.focus()
+        e.preventDefault()
+      }
+    }
+
+    return (
+      <li
+        className={s.switcherOption}
+        id={generateSwitcherOptionIdFromProduct(product)}
+        key={product.code}
+      >
+        <a
+          aria-current={isCurrentProduct ? 'page' : undefined}
+          className={classNames(
+            s.switcherOptionAnchor,
+            s.switcherOptionContainer
+          )}
+          href={product.url}
+          onKeyDown={handleAnchorKeyDown}
+        >
+          <ProductIcon product={product.code} />
+          <span>{product.name}</span>
+        </a>
+      </li>
+    )
+  }
+
+  /**
+   * I _think_ we want the containing element to be a nav, currently clashes with other
+   * styles so not using that element just yet
+   */
+  return (
+    <div className={s.productSwitcher} ref={productChooserRef}>
+      <button
+        aria-controls={OPTION_LIST_ID}
+        aria-expanded={isOpen}
+        aria-labelledby={generateSwitcherOptionIdFromProduct(currentProduct)}
+        className={s.switcherButton}
+        onClick={() => {
+          setIsOpen(!isOpen)
+        }}
+        onKeyDown={handleKeyDown}
+        ref={buttonRef}
+      >
+        <div className={s.switcherOptionContainer}>
```

Comment:
Can we use a span here? `div`'s are not recommended as children of `button` elements. Notice `ProductIcon` returns with wrapping div as well.
<img width="1112" alt="CleanShot 2021-12-13 at 16 56 51@2x" src="https://user-images.githubusercontent.com/825855/145895255-9b89550b-9a15-4525-8118-074d3f3f7756.png">

Reply:
Oof! Wasn't even thinking about this. Great catch. 👀

Reply:
Fixed! And double-checked with this validator. 😊